pkg/core/resources/apis/meshmultizoneservice/api/v1alpha1/meshmultizoneservice.go

```diff
@@ -21,14 +21,15 @@ type MeshMultiZoneService struct {
 	Selector Selector `json:"selector"`
 	// Ports is a list of ports from selected MeshServices
 	// +kubebuilder:validation:MinItems=1
-	Ports []Port `json:"ports,omitempty"`
+	Ports []Port `json:"ports"`
 }
 
 type Port struct {
-	Name string `json:"name,omitempty"`
-	Port uint32 `json:"port"`
+	Name *string `json:"name,omitempty"`
+	Port uint32  `json:"port"`
+	// +kubebuilder:validation:Optional
 	// +kubebuilder:default=tcp
-	AppProtocol core_mesh.Protocol `json:"appProtocol,omitempty"`
+	AppProtocol core_mesh.Protocol `json:"appProtocol"`
```

Comment:
we can also add an enum annotation here, the same way we have on MES:
```go
// +kubebuilder:validation:Enum=tcp;grpc;http;http2
Protocol core_mesh.Protocol `json:"protocol"`
```

Reply:
we agreed on slack to not add enums yet, will create an issue for this